### ISSUE TITLE
Unable to fetch output from previous task

### ISSUE THREAD

**Issue by jatinkumarg** [argoproj/argo-workflows#9475]
### Checklist

- [x] Double-checked my configuration.
- [x] Tested using the latest version.
- [x] Used the Emissary executor.

------------------------------

I have a workflow with dag with two tasks. From the first task, I'm expecting the output `630e8f72f2718e75edb90015` but instead getting this error while passing the output result to the second task:

```
job-2174559: /bin/sh: process: not found
job-2174559: /bin/sh: src/run.ts: Permission denied
job-2174559: /bin/sh: 630e8f72f2718e75edb90015: not found
````

First I thought its access issue so granted admin privileges but still no luck. Could anyone please help me with this? I'm stuck for days now.

```
kubectl create rolebinding default-admin --clusterrole=admin --serviceaccount=argo:default -n argo
```

**Comment by tim-sendible** [comment]
Can you share your example workflow? Not sure anyone's going to be able to help based on what you've written here.

What you've said here simply implies that your container doesn't have a shell. This would be unrelated to workflows itself.

**Comment by jatinkumarg** [comment]
I don't think I'll be able to show my whole workflow as it calls a private image so showing the passed params won't be ethical here.

In simple words, when the first task finished and outputs the result to second task, second tasks is unable to fetch the output result from first task and outputs this permission denied error. 

It was working fine before but ever since I updated argo version on local to v3.3.9 I have been getting same issue. So far I have tried downlograding, re-creating minikube cluster from scratch and even tried this basic operation with older argo versions (3.3.8, 3.3.7, 3.3.6, 3.3.5) but getting same issue.
